cache/cache_rep.h

```diff
@@ -0,0 +1,53 @@
+//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+//
+// Copyright (c) 2011 The LevelDB Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file. See the AUTHORS file for names of contributors.
+
+#pragma once
+
+#include <atomic>
+#include <cstddef>
+#include <cstdint>
+#include <memory>
+#include <mutex>
+#include <vector>
+
+#include "rocksdb/cache.h"
+#include "rocksdb/slice.h"
+
+// [DO-NOT-COMMIT] [A header inclusion issue][A temporary hack]
+// Including the following two dependencies in cache_rep.h leads to a wall of compiler errors when including cache_rep.h in write_buffer_manager.h
+// It is suspected to be circular dependency issue
+// Therefore I commented these two dependencies out and redefine the const from these two dependencies in this file [DO-NOT-COMMIT] 
+// #include "table/block_based/block_based_table_reader.h"
+// #include "util/coding.h"
+
+namespace ROCKSDB_NAMESPACE {
+class CacheRep {
+ public:
+  // The key will be longer than keys for blocks in SST files so they won't
+  // conflict.
+
+  // [DO-NOT-COMMIT][A header inclusion issue]
+  // The following four lines (including the two commented lines) are for adapting to the [A temporary hack] mentioned above
+  // static const std::size_t kCacheKeyPrefixSize =
+  //     BlockBasedTable::kMaxCacheKeyPrefixSize + kMaxVarint64Length;
+  static const std::size_t kMaxVarint64Length = 10;
```

Comment:
why we need to redefine them here? In my impression, they are already defined in coding.h

Reply:
Great question - it is actually due to the reason in the note I left in line 22 in this PR's comment section. This is a temporary hack that is not intended to be committed (therefore this PR is marked "early review"). The reason why this hack exists is that - when including the header where this constant is defined (i.e #include "table/block_based/block_based_table_reader.h") in this file (i.e, cache_rep.h) and including this file (i.e, cache_rep.h) in write_buffer_manager.h, which is what I have to do, there is a wall of compiler errors, indicating a circular dependency issue that I am still troubleshooting for.  Therefore I added the hack here just to bypass this particular issue for reviewing purpose.

I left a note in line 22 and some code comments (e.g, line 22 - 27, line 35 - 38) for this issue. But reading your question makes me realize the way I flag out this hack to potential reviewers is not enough :) Let me flag this issue out more explicitly in the PR comment section. Thank you for you question!

Reply:
My early impression is that kCacheKeyPrefixSize can live in cache_rep.h. I don't think we need to spend too much time polishing these details, as I'm working on a replacement schema for cache keys, that would make them all 16 bytes. (Details hopefully in next day or so.) We probably wouldn't switch over to the new schema all at once, so the current schema might need to stick around for a while.

But work toward unifying all the code generating cache keys under a shared API is moving in the right direction, and I'm not really to a point of getting merge/rebase conflicts with you.

Reply:
Thank you for providing the context on cache key unification! I am aware of this unification before starting this task and very excited about it :)

In terms of kCacheKeyPrefixSize, I will still refer to the coding.h since the "suspected-to-be" circular dependency issue mentioned above is made irrelevant to this PR in a non-hack way after a minor fix I did just now. But thank you for the input!

@zhichao-cao tagged you here just for FYI for the update on the "suspected-to-be" circular dependency issue mentioned in my reply to your comment above. Now this issue is made irrelevant to my PR in a non-hack way, see my latest commit https://github.com/facebook/rocksdb/pull/8506/commits/e5ddc41799a616e3d35946b8df9e70f9e0b83e3a) for more information if you are interested.